src/ray/raylet/node_manager.cc

```diff
@@ -912,96 +820,114 @@ void NodeManager::HandleDisconnectedActor(const ActorID &actor_id, bool was_loca
   if (was_local) {
     // If the actor was local, immediately update the state in actor registry.
     // So if we receive any actor tasks before we receive GCS notification,
-    // these tasks can be correctly routed to the `MethodsWaitingForActorCreation` queue,
-    // instead of being assigned to the dead actor.
+    // these tasks can be correctly routed to the `MethodsWaitingForActorCreation`
+    // queue, instead of being assigned to the dead actor.
     HandleActorStateTransition(actor_id, ActorRegistration(new_actor_data));
   }
   ray::gcs::ActorTable::WriteCallback failure_callback = nullptr;
   if (was_local) {
     failure_callback = [](gcs::AsyncGcsClient *client, const ActorID &id,
                           const ActorTableData &data) {
-      // If the disconnected actor was local, only this node will try to update actor
-      // state. So the update shouldn't fail.
+      // If the disconnected actor was local, only this node will try to update
+      // actor state. So the update shouldn't fail.
       RAY_LOG(FATAL) << "Failed to update state for actor " << id;
     };
   }
   PublishActorStateTransition(actor_id, new_actor_data, failure_callback);
 }
 
-void NodeManager::HandleWorkerAvailable(
-    const std::shared_ptr<LocalClientConnection> &client) {
-  std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
-  RAY_CHECK(worker);
-  // If the worker was assigned a task, mark it as finished.
-  if (!worker->GetAssignedTaskId().IsNil()) {
-    FinishAssignedTask(*worker);
+/// Handle a `GetTask` request.
+void NodeManager::HandleGetTaskRequest(const rpc::GetTaskRequest &request,
+                                       rpc::GetTaskReply *reply,
+                                       rpc::SendReplyCallback send_reply_callback) {
+  const WorkerID worker_id = WorkerID::FromBinary(request.worker_id());
+  std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(worker_id);
+  RAY_LOG(DEBUG) << "Received a GetTaskRequest, worker id " << worker_id << " pid "
+                 << worker->Pid();
+  if (worker->IsBeingKilled()) {
+    send_reply_callback(Status::Invalid("WorkerBeingKilled"), nullptr, nullptr);
+    return;
   }
+  RAY_CHECK(worker && !worker->UsePush());
```

Comment:
I would move the check for `worker` out to an above line, since it would already trigger a segfault at the `worker->IsBeingKilled()` line above.

Reply:
Good catch, thanks. I add worker->IsBeingKilled() sentences after this RAY_CHECK and forgot it.